src/Device/Driver/GDL90/Driver.cpp

```
@@ -0,0 +1,436 @@
+      /* keep current partial frame */
+      if (buffer.size() > 4096)
+        buffer.erase(buffer.begin(), buffer.end() - 4096);
+      break;
+    }
+
+    const std::span<const uint8_t> frame{
+      buffer.data() + 1, std::size_t(end - buffer.begin() - 1)};
+
+    buffer.erase(buffer.begin(), end + 1);
+
+    if (frame.size() < 4)
```

Cppcheck (2.20.0)

[error] Using object that points to member variable 'buffer' that may be invalid.

(invalidContainer)